src/plugins/hiddenRows/test/publicAPI.e2e.js

```diff
@@ -13,34 +13,126 @@ describe('HiddenRows', () => {
-        hiddenRows: true,
+    describe('hideRow()', () => {
+      it('should hide row by passing the visual row index', () => {
+        handsontable({
+          data: Handsontable.helper.createSpreadsheetData(3, 3),
+          hiddenRows: true,
+        });
+
+        expect(getCell(1, 0).innerText).toBe('A2');
+
+        getPlugin('hiddenRows').hideRow(1);
+        render();
+
+        expect(getCell(1, 0)).toBe(null);
       });
+    });
+
+    describe('showRow()', () => {
+      it('should show row by passing the visual row index', () => {
+        handsontable({
+          data: Handsontable.helper.createSpreadsheetData(3, 3),
+          hiddenRows: {
+            rows: [1],
+          },
+        });
+
+        expect(getCell(1, 0)).toBe(null);
+
+        getPlugin('hiddenRows').showRow(1);
+        render();
+
+        expect(getCell(1, 0).innerText).toBe('A2');
+      });
+    });
+
+    describe('isHidden()', () => {
+      it('should return `true` for hidden row', () => {
+        handsontable({
+          data: Handsontable.helper.createSpreadsheetData(3, 3),
+          hiddenRows: {
+            rows: [1],
+          },
+        });
+
+        const plugin = getPlugin('hiddenRows');
+
+        expect(plugin.isHidden(0)).toBe(false);
+        expect(plugin.isHidden(1)).toBe(true);
+        expect(plugin.isHidden(2)).toBe(false);
 
-      expect(getCell(1, 0).innerText).toBe('A2');
+        getPlugin('hiddenRows').showRow(1);
+        render();
 
-      getPlugin('hiddenRows').hideRow(1);
-      render();
+        expect(plugin.isHidden(0)).toBe(false);
+        expect(plugin.isHidden(1)).toBe(false);
+        expect(plugin.isHidden(2)).toBe(false);
 
-      expect(getCell(1, 0)).toBe(null);
+        getPlugin('hiddenRows').hideRow(2);
+        render();
+
+        expect(plugin.isHidden(0)).toBe(false);
+        expect(plugin.isHidden(1)).toBe(false);
+        expect(plugin.isHidden(2)).toBe(true);
+      });
     });
 
-    it('should show row after calling the showrow method', () => {
-      handsontable({
-        data: Handsontable.helper.createSpreadsheetData(3, 3),
-        hiddenRows: {
-          rows: [1],
-        },
+    describe('getHiddenRows()', () => {
+      it('should return collection of hidden visual row indexes', () => {
+        handsontable({
+          data: Handsontable.helper.createSpreadsheetData(3, 3),
+          hiddenRows: {
+            rows: [1],
+          },
+        });
+
+        const plugin = getPlugin('hiddenRows');
+
+        expect(plugin.getHiddenRows()).toEqual([1]);
+
+        getPlugin('hiddenRows').showRow(1);
+        render();
+
+        expect(plugin.getHiddenRows()).toEqual([]);
+
+        getPlugin('hiddenRows').hideRows([0, 2]);
+        render();
+
+        expect(plugin.getHiddenRows()).toEqual([0, 2]);
       });
 
-      expect(getCell(1, 0)).toBe(null);
+      it('should return correct visual indexes when some rows are sorted ' +
```

Comment:
Maybe using `rowIndexMapper.setIndexesSequence` will make this test more general?

Reply:
Fixed within https://github.com/handsontable/handsontable/pull/6887/commits/12655780da5eb57c2f11e5a750c701304e8238cd.